service/hook/gogs/gogs.go

```diff
@@ -104,35 +126,44 @@ func (hp HookProvider) TransformRequest(r *http.Request) hookCommon.TransformRes
 	contentType, eventID, err := detectContentTypeAndEventID(r.Header)
 	if err != nil {
 		return hookCommon.TransformResultModel{
-			Error: fmt.Errorf("Issue with Headers: %s", err),
+			Error:      fmt.Errorf("Issue with Headers: %s", err),
+			ShouldSkip: true,
 		}
 	}
 
-	if contentType != "application/json" {
+	if contentType != hookCommon.ContentTypeApplicationJSON {
 		return hookCommon.TransformResultModel{
 			Error: fmt.Errorf("Content-Type is not supported: %s", contentType),
 		}
 	}
 
-	if eventID != "push" {
-		// Unsupported Event
-		return hookCommon.TransformResultModel{
-			Error: fmt.Errorf("Unsupported Webhook event: %s", eventID),
-		}
-	}
-
 	if r.Body == nil {
 		return hookCommon.TransformResultModel{
 			Error: fmt.Errorf("Failed to read content of request body: no or empty request body"),
 		}
 	}
 
-	// code push
-	var codePushEvent CodePushEventModel
-	if contentType == "application/json" {
-		if err := json.NewDecoder(r.Body).Decode(&codePushEvent); err != nil {
+	if eventID == pushEventID {
+		var pushEvent PushEventModel
+		if err := json.NewDecoder(r.Body).Decode(&pushEvent); err != nil {
+			return hookCommon.TransformResultModel{Error: fmt.Errorf("Failed to parse request body: %s", err)}
+		}
+
+		return transformPushEvent(pushEvent)
+
+	} else if eventID == tagEventID {
```

Comment:
I think this is not enough, I'm almost sure that when you create a new branch that's a `create` event too.